ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/datetime/FormatTimestamp.java

```diff
@@ -0,0 +1,95 @@
+    category = FunctionCategory.DATE_TIME,
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "Converts a TIMESTAMP value into the string representation of the timestamp"
+        + " in the given format."
+)
+public class FormatTimestamp {
+
+  private final LoadingCache<String, DateTimeFormatter> formatters =
+      CacheBuilder.newBuilder()
+          .maximumSize(1000)
+          .build(CacheLoader.from(DateTimeFormatter::ofPattern));
+
+  @Udf(description = "Converts a TIMESTAMP value into the"
+      + " string representation of the timestamp in the given format. Single quotes in the"
+      + " timestamp format can be escaped with '', for example: 'yyyy-MM-dd''T''HH:mm:ssX'"
+      + " The system default time zone is used when no time zone is explicitly provided."
+      + " The format pattern should be in the format expected"
+      + " by java.time.format.DateTimeFormatter")
+  public String formatTimestamp(
+      @UdfParameter(
+          description = "TIMESTAMP value.") final Timestamp timestamp,
+      @UdfParameter(
+          description = "The format pattern should be in the format expected by"
+              + " java.time.format.DateTimeFormatter.") final String formatPattern) {
+    try {
+      final DateTimeFormatter formatter = formatters.get(formatPattern);
+      return timestamp.toInstant()
+          .atZone(ZoneId.systemDefault())
+          .format(formatter);
+    } catch (final ExecutionException | RuntimeException e) {
+      throw new KsqlFunctionException("Failed to format timestamp " + timestamp
+          + " with formatter '" + formatPattern
+          + "': " + e.getMessage(), e);
+    }
+  }
+
+  @Udf(description = "Converts a BIGINT millisecond timestamp value into the"
+      + " string representation of the timestamp in the given format. Single quotes in the"
+      + " timestamp format can be escaped with '', for example: 'yyyy-MM-dd''T''HH:mm:ssX'")
+  public String formatTimestamp(
+      @UdfParameter(
+          description = "TIMESTAMP value.") final Timestamp timestamp,
+      @UdfParameter(
+          description = "The format pattern should be in the format expected by"
+              + " java.time.format.DateTimeFormatter.") final String formatPattern,
+      @UdfParameter(
+          description =  " timeZone is a java.util.TimeZone ID format, for example: \"UTC\","
+              + " \"America/Los_Angeles\", \"PST\", \"Europe/London\"") final String timeZone) {
+    try {
+      final DateTimeFormatter formatter = formatters.get(formatPattern);
+      final ZoneId zoneId = ZoneId.of(timeZone);
+      return timestamp.toInstant()
+          .atZone(zoneId)
+          .format(formatter);
+    } catch (final ExecutionException | RuntimeException e) {
+      throw new KsqlFunctionException("Failed to format timestamp " + timestamp
+          + " at timeZone '" + timeZone + "' with formatter '" + formatPattern
+          + "': " + e.getMessage(), e);
+    }
```

Comment:
All this code is similar to the other formatTimestamp except for the timeZone. Would it be good to have a method for this?